provider/cluster_rosa_classic_resource.go

```diff
@@ -1844,6 +1891,35 @@ func proxyValidators() []tfsdk.AttributeValidator {
 	}
 }
 
+func rootVolumeValidators() []tfsdk.AttributeValidator {
+	return []tfsdk.AttributeValidator{
+		&common.AttributeValidator{
+			Desc: "Validate root volume attributes",
+			Validator: func(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+				state := new(MachineTypeRootVolume)
+				diag := req.Config.GetAttribute(ctx, req.AttributePath, state)
+				if diag.HasError() {
+					// No attribute to validate
+					return
+				}
+				errSum := "Invalid root volume's attribute assignment"
+				var rootVolumeSize int64
+
+				if state.Size != 0 {
+					rootVolumeSize = state.Size
+				}
+
+				// Validate provided size
+				// TODO: do better
+				if rootVolumeSize < 0 {
+					resp.Diagnostics.AddError(errSum, "Expected root volume size to be greater than 0")
+					return
+				}
+			},
+		},
+	}
+}
+
 func propertiesValidators() []tfsdk.AttributeValidator {
```

Comment:
You should also add the assignment of those properties in the "Create" function
Also, if this attribute (compute_machine_type_state) can be updated and modified, please add the support to the "update" function.